core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go

```diff
@@ -314,32 +307,51 @@ func (amc *AssetManagementContract) UnlockAsset(ctx contractapi.TransactionConte
             err = ctx.GetStub().SetEvent("UnlockAsset", contractInfoBytes)
         }
         if err != nil {
-            log.Warn("Unable to set 'UnlockAsset' event")
-            log.Warn(err.Error())
+	    logWarnings("Unable to set 'UnlockAsset' event", err.Error())
         }
     }
     return retVal, err
 }
 
 func (amc *AssetManagementContract) UnlockFungibleAsset(ctx contractapi.TransactionContextInterface, contractId string) (bool, error) {
     if len(contractId) == 0 {
-        log.Error("empty contract id")
-        return false, fmt.Errorf("empty contract id")
+        return false, logThenErrorf("empty contract id")
     }
 
     // The below 'SetEvent' should be the last in a given transaction (if this function is being called by another), otherwise it will be overridden
     retVal, err := amc.assetManagement.UnlockFungibleAsset(ctx.GetStub(), contractId)
     if retVal && err == nil {
-        contractInfo := &common.AssetContractHTLC{
+        contractInfo := &common.FungibleAssetContractHTLC{
             ContractId: contractId,
         }
         contractInfoBytes, err := proto.Marshal(contractInfo)
         if err == nil {
             err = ctx.GetStub().SetEvent("UnlockFungibleAsset", contractInfoBytes)
         }
         if err != nil {
-            log.Warn("Unable to set 'UnlockFungibleAsset' event")
-            log.Warn(err.Error())
+	    logWarnings("Unable to set 'UnlockFungibleAsset' event", err.Error())
+        }
+    }
+    return retVal, err
+}
+
+func (amc *AssetManagementContract) UnlockAssetUsingContractId(ctx contractapi.TransactionContextInterface, contractId string) (bool, error) {
+    if len(contractId) == 0 {
+        return false, logThenErrorf("empty contract id")
+    }
+
+    // The below 'SetEvent' should be the last in a given transaction (if this function is being called by another), otherwise it will be overridden
+    retVal, err := amc.assetManagement.UnlockAssetUsingContractId(ctx.GetStub(), contractId)
+    if retVal && err == nil {
+        contractInfo := &common.AssetContractHTLC{
+            ContractId: contractId,
+        }
+        contractInfoBytes, err := proto.Marshal(contractInfo)
+        if err == nil {
+            err = ctx.GetStub().SetEvent("UnlockAssetUsingContractId", contractInfoBytes)
```

Comment:
We can just set this event name to `UnlockAsset`. The event receiver will see the same structure regardless of whether it contains a contract id or not, and can accordingly perform suitable actions.

Reply:
Done @VRamakrishna !